Makefile

```diff
@@ -56,7 +56,9 @@ prefix_install: installables
 	mkdir -p "$(PREFIX)/Frameworks" "$(PREFIX)/bin"
 	rsync -a --delete "$(TEMPORARY_FOLDER)$(FRAMEWORKS_FOLDER)/$(OUTPUT_FRAMEWORK)" "$(PREFIX)/Frameworks/"
 	cp -f "$(TEMPORARY_FOLDER)$(BINARIES_FOLDER)/carthage" "$(PREFIX)/bin/"
-	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks/$(OUTPUT_FRAMEWORK)/Versions/Current/Frameworks/"  "$(PREFIX)/bin/carthage"
+	install_name_tool -delete_rpath "@executable_path/../Frameworks" "$(PREFIX)/bin/carthage" # Avoid duplication of "@executable_path/../Frameworks"
+	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks" "$(PREFIX)/bin/carthage"
```

Comment:
Unnecessary entries remain in the execution of `carthage`, so I am uncomfortable personally.
All dyld issues are due to the fact that `rpath` has unnecessary entries for execution, and we solved those issues by minimizing `rpath` in SwiftLint.
I have written a [comment](https://github.com/realm/SwiftLint/pull/1624#issuecomment-308686852) on `rpath` which SwiftLint requires at the minimum and it should be helpful in carthage as well.